src/bridge/EspressoTEEVerifier.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {
+    V3QuoteVerifier
+} from "@automata-network/dcap-attestation/contracts/verifiers/V3QuoteVerifier.sol";
+import {BELE} from "@automata-network/dcap-attestation/contracts/utils/BELE.sol";
+import {Header} from "@automata-network/dcap-attestation/contracts/types/CommonStruct.sol";
+import {
+    IQuoteVerifier
+} from "@automata-network/dcap-attestation/contracts/interfaces/IQuoteVerifier.sol";
+import {
+    HEADER_LENGTH,
+    ENCLAVE_REPORT_LENGTH
+} from "@automata-network/dcap-attestation/contracts/types/Constants.sol";
+import {EnclaveReport} from "@automata-network/dcap-attestation/contracts/types/V3Structs.sol";
+/**
+ *
+ * @title  Verifies quotes from the TEE and attests on-chain
+ * @notice Contains the logic to verify a quote from the TEE and attest on-chain. It uses the V3QuoteVerifier contract
+ *         to verify the quote. Along with some additional verification logic.
+ */
+
+contract EspressoTEEVerifier is V3QuoteVerifier {
+    constructor(address _router) V3QuoteVerifier(_router) {}
+
+    /**
+        @notice Verify a quote from the TEE and attest on-chain
+        @param rawQuote The quote from the TEE
+        @return success True if the quote was verified and attested on-chain
+     */
+    function verify(bytes calldata rawQuote) external view returns (bool success) {
+        // Parse the header
+        Header memory header = _parseQuoteHeader(rawQuote);
+
+        if (header.version != 3) {
+            return false;
+        }
+
+        (success, ) = this.verifyQuote(header, rawQuote);
+        if (!success) {
+            return false;
+        }
+
+        //  Parse enclave quote
+        uint256 offset = HEADER_LENGTH + ENCLAVE_REPORT_LENGTH;
+        EnclaveReport memory localReport;
+        (success, localReport) = parseEnclaveReport(rawQuote[HEADER_LENGTH:offset]);
```

Comment:
Getting up to speed here - what's the purpose of parsing the enclave report?
What's different between that and verifying the quote? 

From looking at Automata's `V3QuoteVerifier`, the `verifyQuote` function calls [`parseQuote`](https://github.com/automata-network/automata-dcap-attestation/blob/main/contracts/verifiers/V3QuoteVerifier.sol#L39) which also calls [`parseEnclaveReport`](https://github.com/automata-network/automata-dcap-attestation/blob/main/contracts/verifiers/V3QuoteVerifier.sol#L66)

Reply:
We will add more verification logic in the upcoming PR's so we need the enclave report for that